extensions/wikia/SEOTweaks/SEOTweaksHooksHelper.class.php

```diff
@@ -361,4 +361,40 @@ public static function onLinkEnd( $skin, Title $target, array $options, &$text,
 
 		return true;
 	}
+
+	protected static function findLanguagePath( array $parsedUrl ): string {
+		if ( !array_key_exists( 'path', $parsedUrl ) ) {
+			return '';
+		}
+
+		$path = $parsedUrl['path'];
+		if ( strlen( $path ) === 0 ) {
+			return '';
+		}
+
+		$langCode = explode( '/', $path )[1];
+		$languages = Language::getLanguageNames();
+		if ( isset( $languages[$langCode] ) ) {
+			return '/' . $langCode;
+		}
+
+		return '';
+	}
+
+	public static function onLinkerMakeExternalLink( string &$url, string &$text, bool &$link, array &$attribs ): bool {
+		$parsed = parse_url( $url );
+
+		if ( $parsed !== false ) {
+			$host = $parsed['host'];
+			$path = self::findLanguagePath( $parsed );
+			$city_id = WikiFactory::DomainToID( wfNormalizeHost( $host . $path ) );
+			if ( $city_id ) {
+				$primaryCityUrl = parse_url( WikiFactory::cityIDtoDomain( $city_id ) );
+				$parsed['host'] = $primaryCityUrl['host'];
```

Comment:
there are few cases for which it does not work
1. when the old domain is de.starwars.wikia.com and the primary is starwars.wikia.com/de, you're not adding the de to the language path
2. if the primary domain is something.fandom.com and someone used the English alias something.fandom.com/en, you're not stripping the en part
3.  Should we use `getLocalEnvURL`?
4. you could really use the unit tests here :)